pyiron_workflow/snippets/files.py

```diff
@@ -113,3 +115,33 @@ def is_file(self):
 
     def delete(self):
         self.path.unlink()
+
+    def _resolve_directory_and_path(
+        self,
+        file_name: str,
+        directory: DirectoryObject | str | None=None,
+        default_directory: str=".",
+    ):
+        """
+        Internal routine to separate the file name and the directory in case
+        file name is given in absolute path etc.
+        """
+        path = Path(file_name)
+        file_name = path.name
+        if path.is_absolute():
+            # If absolute path, take that of new_file_name regardless of the
+            # name of directory
```

Comment:
In fact, I think you can actually reduce the conditional nesting by one level by introducing

```python
absolute_parent = str(path.parent) if path.is_absolute() else None
```

And then proceeding to parsing non-None `directory` into a `DirectoryObject`, raising an error if both are non-None and don't match, or using the default if both are None.

Reply:
but is it a realistic situation that someone sets an absolute path for the file and defines the directory at the same time?

Reply:
Likely? No. Realistic? Sure. It could be a novice who doesn't even know what an absolute path is and just put "/" at the start of their path because they think they need it in order to get their file name to append onto their directory path, or it could be someone who's in a rush and copying and pasting names from one place to another and happens to mis-copy/paste something absolute when they didn't mean to.

I don't see that handling the case adds undue strain on our code base, so if we expose the possibility of a  a crazy input combination we should lock it down cleanly.

Reply:
Since I don't know what kind of situation it is that the user sets an absolute path for the file name and specifies a directory, I decided to make it raise an error as soon as that happens, regardless of compatibility between the two. There's probably some logic in every case, but my biggest concern is something like this: `FileObject("/home/my_file.txt", "/home")`, in which case, I don't know which one of the following options was meant to be:

- `FileObject("home/my_file.txt", "/home")`, i.e. the file is `/home/home/my_file.txt`
- `FileObject("my_file.txt", "/home")`, i.e. the file is `/home/my_file.txt`

In both cases, or in any other possible combinations, I don't think it goes beyond silly redundancy and/or typo, so I don't think it is irrational to raise an error.